lib/view_component/base.rb

```diff
@@ -241,6 +241,14 @@ def collection_parameter_name
         (@with_collection_parameter || name.demodulize.underscore.chomp("_component")).to_sym
       end
 
+      def collection_counter_parameter_name
```

Comment:
I _believe_ this should be something like `define_singleton_method`, like above:

https://github.com/github/view_component/pull/306/files/302defcc84a73ac0b55f9a6b66f8b6384f6703c7..1c168aa737bfbdcd357777e7bce9ebefbb2ef3c8#diff-08fe1c0f29a6308ad005c21f4adcfd64R186

(My apologies for not sharing this example case in my earlier comment ❤️ )

Reply:
No need to apologise 😄 

I think I committed my second attempt as you were writing this comment. I already pushed a fix after I reread everything again 👍 Here is the commit 7c0676b87e063fe214a2605cb41bc708ea7778ea